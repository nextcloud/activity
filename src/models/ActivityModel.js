/**
 * @copyright Copyright (c) 2021 Louis Chemineau <louis@chmn.me>
 *
 * @author Louis Chemineau <louis@chmn.me>
 *
 * @license GNU AGPL version 3 or any later version
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 *
 */

import moment from '@nextcloud/moment'

export default class ActivityModel {

	_activity;

	/**
	 * Create the activity object
	 *
	 * @param {Object} rawActivity the activity object from the ocs response
	 */
	constructor(rawActivity) {
		if (typeof rawActivity !== 'object') {
			throw new Error('Received activity data is not an object.')
		}

		// Sanity checks
		if (typeof rawActivity.activity_id !== 'number') {
			throw new Error('The activity_id argument is not a valid string')
		}
		if (typeof rawActivity.type !== 'string' || rawActivity.type.trim() === '') {
			throw new Error('The activity_id argument is not a valid string')
		}
		if (typeof rawActivity.subject !== 'string' || rawActivity.subject.trim() === '') {
			throw new Error('The activity_id argument is not a valid string')
		}
		if (typeof rawActivity.icon !== 'string' || rawActivity.icon.trim() === '') {
			throw new Error('The activity_id argument is not a valid string')
		}
		if (typeof rawActivity.datetime !== 'string' || rawActivity.datetime.trim() === '') {
			throw new Error('The activity_id argument is not a valid string')
		}

		// store state
		this._activity = rawActivity
	}

	/**
	 * get the activity id
	 *
	 * @returns {int}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get id() {
		return this._activity.id
	}

	/**
	 * Get the activity app
	 *
	 * @returns {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get app() {
		return this._activity.app
	}

	/**
	 * Get the activity type
	 *
	 * @returns {int}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get type() {
		return this._activity.type
	}

	/**
	 * Get the activity user
	 *
	 * @returns {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get user() {
		return this._activity.user
	}

	/**
	 * Get the activity subject
	 *
	 * @returns {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get subject() {
		return this._activity.subject
	}

	/**
	 * Get the activity subject_rich template
	 *
	 * @returns {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get subjectRichTemplate() {
		return this._activity.subject_rich[0]
	}

	/**
	 * Get the activity subject_rich objects
	 *
	 * @returns {Object.<string, RichObject>}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get subjectRichObjects() {
		if (Array.isArray(this._activity.subject_rich[1])) {
			return {}
		}

		return this._activity.subject_rich[1]
	}

	/**
	 * Get the activity message
	 *
	 * @returns {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get message() {
		return this._activity.message
	}

	/**
	 * Get the activity message_rich template
	 *
	 * @returns {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get messageRichTemplate() {
		return this._activity.message_rich[0]
	}

	/**
	 * Get the activity message_rich objects
	 *
	 * @returns {Object.<string, RichObject>}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get messageRichObjects() {
		if (Array.isArray(this._activity.message_rich[1])) {
			return {}
		}

		return this._activity.message_rich[1]
	}

	/**
	 * Get the object_type
	 *
	 * @returns {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get objectType() {
		return this._activity.object_type
	}

	/**
	 * Get the activity object_id
	 *
	 * @returns {int}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get objectId() {
		return this._activity.object_id
	}

	/**
	 * Get the activity object_name
	 *
	 * @returns {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get objectName() {
		return this._activity.object_name
	}

	/**
	 * Get the activity link
	 *
	 * @returns {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get link() {
		return this._activity.link
	}

	/**
	 * Get the activity icon
	 *
	 * @returns {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get icon() {
		return this._activity.icon
	}

	/**
	 * Get the activity datetime
	 *
	 * @returns {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get datetime() {
		return this._activity.datetime
	}

	/**
	 * Get the activity formatted date from the current date
	 *
	 * @returns {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get dateFromNow() {
		return moment(this._activity.datetime).fromNow()
	}

	/**
	 * Get the activity formatted datetime
	 *
	 * @returns {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get formattedDate() {
		return moment(this._activity.datetime).format('LLL')
	}

	/**
	 * Get the activity timestamp
	 *
	 * @returns {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get timestamp() {
		return moment(this._activity.datetime).unix()
	}

}
