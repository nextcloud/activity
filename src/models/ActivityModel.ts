/**
 * SPDX-FileCopyrightText: 2021 Nextcloud GmbH and Nextcloud contributors
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

import type { IPreview, IRawActivity, IRichObject } from './types'
import moment from '@nextcloud/moment'

export default class ActivityModel {

	_activity: IRawActivity

	/**
	 * Create the activity object
	 *
	 * @param {IRawActivity} rawActivity the activity object from the ocs response
	 */
	constructor(rawActivity: IRawActivity) {
		if (typeof rawActivity !== 'object') {
			throw new Error('Received activity data is not an object.')
		}

		// Sanity checks
		if (typeof rawActivity.activity_id !== 'number') {
			throw new Error('The activity_id argument is not a valid string')
		}
		if (typeof rawActivity.type !== 'string' || rawActivity.type.trim() === '') {
			throw new Error('The activity_id argument is not a valid string')
		}
		if (typeof rawActivity.subject !== 'string' || rawActivity.subject.trim() === '') {
			throw new Error('The activity_id argument is not a valid string')
		}
		if (typeof rawActivity.icon !== 'string' || rawActivity.icon.trim() === '') {
			throw new Error('The activity_id argument is not a valid string')
		}
		if (typeof rawActivity.datetime !== 'string' || rawActivity.datetime.trim() === '') {
			throw new Error('The activity_id argument is not a valid string')
		}

		// store state
		this._activity = rawActivity
	}

	/**
	 * get the activity id
	 *
	 * @return {number}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get id(): number {
		return this._activity.activity_id
	}

	/**
	 * Get the app causing the activity
	 *
	 * @return {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get app(): string {
		return this._activity.app
	}

	/**
	 * Get the activity type
	 *
	 * @return {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get type(): string {
		return this._activity.type
	}

	/**
	 * Get the user ID of the user causing the activity or affected by the activity
	 *
	 * @return {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get user(): string {
		return this._activity.user
	}

	/**
	 * Get the activity subject
	 *
	 * @return {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get subject(): string {
		return this._activity.subject
	}

	/**
	 * Get the activity subject_rich template
	 *
	 * @return {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get subjectRichTemplate(): string {
		return this._activity.subject_rich[0]
	}

	/**
	 * Get the activity subject_rich objects
	 *
	 * @return {Record<string, IRichObject>}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get subjectRichObjects(): Record<string, IRichObject> {
		if (Array.isArray(this._activity.subject_rich[1])) {
			return {}
		}

		return this._activity.subject_rich[1] as Record<string, IRichObject>
	}

	/**
	 * Get the activity message
	 *
	 * @return {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get message(): string {
		return this._activity.message
	}

	/**
	 * Get the activity message_rich template
	 *
	 * @return {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get messageRichTemplate(): string {
		return this._activity.message_rich[0]
	}

	/**
	 * Get the activity message_rich objects
	 *
	 * @return {Record<string, IRichObject>}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get messageRichObjects(): Record<string, IRichObject> {
		if (!Array.isArray(this._activity.message_rich[1])) {
			return {}
		}

		return this._activity.message_rich[1] as Record<string, IRichObject>
	}

	/**
	 * Get the object_type
	 *
	 * @return {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get objectType(): string {
		return this._activity.object_type
	}

	/**
	 * Get the activity object_id
	 *
	 * @return {number}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get objectId(): number {
		return this._activity.object_id
	}

	/**
	 * Get the activity object_name
	 *
	 * @return {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get objectName(): string {
		return this._activity.object_name
	}

	/**
	 * Get the activity link
	 *
	 * @return {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get link(): string {
		return this._activity.link
	}

	/**
	 * Get the activity icon
	 *
	 * @return {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get icon(): string {
		return this._activity.icon
	}

	/**
	 * Get the activity datetime
	 *
	 * @return {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get datetime(): string {
		return this._activity.datetime
	}

	/**
	 * Get the activity formatted date from the current date
	 *
	 * @return {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get dateFromNow(): string {
		return moment(this._activity.datetime).fromNow()
	}

	/**
	 * Get the activity formatted datetime
	 *
	 * @return {string}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get formattedDate(): string {
		return moment(this._activity.datetime).format('LLL')
	}

	/**
	 * Get the activity timestamp
	 *
	 * @return {number}
	 * @readonly
	 * @memberof ActivityModel
	 */
	get timestamp(): number {
		return moment(this._activity.datetime).valueOf()
	}

	/**
	 * Get previews of affected files
	 */
	get previews(): IPreview[] {
		return this._activity.previews ?? []
	}

}
