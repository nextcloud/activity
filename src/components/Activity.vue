<!--
  - @copyright 2021 Louis Chemineau <louis@chmn.me>
  -
  - @author Louis Chemineau <louis@chmn.me>
  -
  - @license GNU AGPL version 3 or any later version
  -
  - This program is free software: you can redistribute it and/or modify
  - it under the terms of the GNU Affero General Public License as
  - published by the Free Software Foundation, either version 3 of the
  - License, or (at your option) any later version.
  -
  - This program is distributed in the hope that it will be useful,
  - but WITHOUT ANY WARRANTY; without even the implied warranty of
  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  - GNU Affero General Public License for more details.
  -
  - You should have received a copy of the GNU Affero General Public License
  - along with this program.  If not, see <http://www.gnu.org/licenses/>.
  -->

<template>
	<li class="activity-entry">
		<Avatar
			class="activity-entry__icon"
			:disable-menu="true"
			:disable-tooltip="true"
			:url="activity.icon"
			:size="15" />

		<div class="activity-entry__content">
			<RichText class="activity-entry__content__subject" :text="subjectText" :arguments="subjectArguments" />
			<RichText class="activity-entry__content__message" :text="messageText" :arguments="messageArguments" />
		</div>

		<span v-tooltip.bottom="activity.formattedDate" class="activity-entry__date">{{ dateFromNow }}</span>
	</li>
</template>

<script>
import Avatar from '@nextcloud/vue/dist/Components/Avatar'
import UserBubble from '@nextcloud/vue/dist/Components/UserBubble'
import RichText from '@juliushaertl/vue-richtext'
import Tooltip from '@nextcloud/vue/dist/Directives/Tooltip'

import ActivityModel from '../models/ActivityModel'

import FileRichArgument from './richArgumentsTypes/FileRichArgument'
import EmailRichArgument from './richArgumentsTypes/EmailRichArgument'
import SystemTagRichArgument from './richArgumentsTypes/SystemTagRichArgument'
import CalendarEventRichArgument from './richArgumentsTypes/CalendarEventRichArgument'
import OpenGraphRichArgument from './richArgumentsTypes/OpenGraphRichArgument'

export default {
	name: 'Activity',
	components: {
		Avatar,
		RichText,
	},
	directives: {
		tooltip: Tooltip,
	},
	props: {
		/**
		 * The activity to render.
		 */
		activity: {
			type: ActivityModel,
			required: true,
		},
	},
	data() {
		return {
			dateFromNow: '',
			dateInterval: 0,
		}
	},
	computed: {
		/**
		 * @returns {String} The activity's messageRichTemplate. Fallback to message if messageRichTemplate does not exists
		 */
		messageText() {
			return this.activity.messageRichTemplate || this.activity.message
		},
		/**
		 * @returns {Object} A map of rich arguments with a Component to build them.
		 */
		messageArguments() {
			return this.mapRichObjectsToRichArguments(this.activity.messageRichObjects)
		},
		/**
		 * @returns {String} The activity's subjectRichTemplate. Fallback to subject if subjectRichTemplate does not exists
		 */
		subjectText() {
			return this.activity.subjectRichTemplate || this.activity.subject
		},
		/**
		 * @returns {Object} A map of rich arguments with a Component to build them.
		 */
		subjectArguments() {
			return this.mapRichObjectsToRichArguments(this.activity.subjectRichObjects)
		},
	},
	created() {
		this.updateDateFromNow()
		this.dateInterval = setInterval(this.updateDateFromNow, 60 * 1000)
	},
	destroyed() {
		clearInterval(this.dateInterval)
	},
	methods: {
		updateDateFromNow() {
			this.dateFromNow = this.activity.dateFromNow
		},
		/**
		 * Map an collection of rich text objects to rich arguments for the RichText component
		 * @arg {Object.<string, RichObject>} richObjects the collection of rich text objects
		 * @returns {Object.<string, RichArgument>}
		 */
		mapRichObjectsToRichArguments(richObjects) {
			const args = {}

			for (const richObjectName in richObjects) {
				args[richObjectName] = this.mapRichObjectToRichArgument(richObjects[richObjectName])
			}

			return args
		},
		/**
		 * Map rich text object to rich argument for the RichText component
		 * @arg {RichObject} richObject the rich text object
		 * @returns {RichArgument}
		 */
		mapRichObjectToRichArgument(richObject) {
			switch (richObject.type) {
			case 'file':
				return {
					component: FileRichArgument,
					props: richObject,
				}
			case 'user':
				return {
					component: UserBubble,
					props: { displayName: richObject.name, user: richObject.id, url: richObject.link },
				}
			case 'group':
				return {
					component: UserBubble,
					props: {
						avatarImage: 'icon-group',
						displayName: richObject.name,
						primary: true,
					},
				}
			case 'email':
				return {
					component: EmailRichArgument,
					props: richObject,
				}
			case 'systemtag':
				return {
					component: SystemTagRichArgument,
					props: richObject,
				}
			case 'opengraph':
				return {
					component: OpenGraphRichArgument,
					props: richObject,
				}
			case 'calendar-event':
				return {
					component: CalendarEventRichArgument,
					props: richObject,
				}
			default:
				return richObject
			}
		},
	},
}
</script>
<style lang="scss" scoped>
.activity-entry {
	display: flex;
	align-items: flex-start;
	width: 100%;
	height: var(--height);
	min-height: 32px;
	padding: 8px 0;

	&__icon {
		opacity: 0.5;
		margin-top: 4px;
		margin-right: 8px;
	}

	&__content {
		display: flex;
		flex-direction: column;
		flex-grow: 1;
		overflow-wrap: break-word;
		white-space: pre-wrap;
		word-break: break-word;
		overflow: hidden;

		&__message {
			color: var(--color-text-lighter);
		}

		::v-deep a {
			font-weight: bold;

			&:hover {
				opacity: 0.7;
				text-decoration: underline;
			}
		}
	}

	&__date {
		color: var(--color-text-lighter);
		margin-left: 5px;
		flex-shrink: 0;
	}
}
</style>
