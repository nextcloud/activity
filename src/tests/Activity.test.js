/**
 * @copyright Copyright (c) 2021 Louis Chemineau <louis@chmn.me>
 *
 * @author Louis Chemineau <louis@chmn.me>
 *
 * @license GNU AGPL version 3 or any later version
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 *
 */
import { render } from '@testing-library/vue'
import Vue from 'vue'

import Activity from '../components/Activity.vue'
import ActivityModel from '../models/ActivityModel'

const wsData = require('../../__mocks__/@nextcloud/activity_ws.json')

jest.useFakeTimers()

const currentDate = new Date('2021-05-10T12:00:00+00:00')
const realDateNow = Date.now

beforeAll(() => {
	global.Date.now = jest.fn(() => currentDate.getTime())
})

afterAll(() => {
	global.Date.now = realDateNow
})

test('Display relative date gets updated every minutes', async() => {
	const { baseElement, getByText } = render(Activity, { props: { activity: new ActivityModel(wsData.ocs.data[1]) } })
	expect(baseElement.textContent).toContain('You renamed  Test file - renamed.md  to  Test file - renamed - looooooooong.md')

	getByText('11 days ago')

	const currentDatePlusOneDay = new Date('2021-05-11T12:00:00+00:00')
	global.Date.now = jest.fn(() => currentDatePlusOneDay.getTime())

	jest.advanceTimersByTime(60 * 1000)
	await Vue.nextTick()
	getByText('12 days ago')

	global.Date.now = jest.fn(() => currentDate.getTime())
})

test('Display correct information for renames', async() => {
	const { baseElement, getByText } = render(Activity, { props: { activity: new ActivityModel(wsData.ocs.data[1]) } })

	expect(baseElement.textContent).toContain('You renamed  Test file - renamed.md  to  Test file - renamed - looooooooong.md')

	const originalFileNameElement = getByText('Test file - renamed.md')
	expect(originalFileNameElement.nodeName).toBe('A')

	const newFileNameElement = getByText('Test file - renamed - looooooooong.md')
	expect(newFileNameElement.nodeName).toBe('A')

	expect(baseElement.innerHTML).toMatchSnapshot()
})

test('Display correct information for comments', async() => {
	const { baseElement } = render(Activity, { props: { activity: new ActivityModel(wsData.ocs.data[4]) } })

	expect(baseElement.textContent).toContain('You commented')

	expect(baseElement.textContent).toContain('A comment')

	expect(baseElement.innerHTML).toMatchSnapshot()
})

test('Display correct information for favorites', async() => {
	const { baseElement } = render(Activity, { props: { activity: new ActivityModel(wsData.ocs.data[5]) } })

	expect(baseElement.textContent).toContain('Added to favorites')

	expect(baseElement.innerHTML).toMatchSnapshot()
})

test('Display correct information for unfavorites', async() => {
	const { baseElement } = render(Activity, { props: { activity: new ActivityModel(wsData.ocs.data[6]) } })

	expect(baseElement.textContent).toContain('Removed from favorites')

	expect(baseElement.innerHTML).toMatchSnapshot()
})

test('Display correct information for changes', async() => {
	const { baseElement, getByText } = render(Activity, { props: { activity: new ActivityModel(wsData.ocs.data[7]) } })

	expect(baseElement.textContent).toContain('You changed  Test file - renamed.md')

	const fileNameElement = getByText('Test file - renamed.md')
	expect(fileNameElement.nodeName).toBe('A')

	expect(baseElement.innerHTML).toMatchSnapshot()
})

test('Display correct information for tags', async() => {
	const { baseElement } = render(Activity, { props: { activity: new ActivityModel(wsData.ocs.data[9]) } })

	expect(baseElement.textContent).toContain('Added system tag tag1')

	expect(baseElement.innerHTML).toMatchSnapshot()
})

test('Display correct information for shares', async() => {
	const { baseElement } = render(Activity, { props: { activity: new ActivityModel(wsData.ocs.data[10]) } })
	expect(baseElement.textContent).toContain('Shared as public link')

	expect(baseElement.innerHTML).toMatchSnapshot()
})

test('Display correct information for moves', async() => {
	const { baseElement, getByText } = render(Activity, { props: { activity: new ActivityModel(wsData.ocs.data[14]) } })

	expect(baseElement.textContent).toContain('You moved  Test file.md  to  Documents')

	const fileNameElement = getByText('Test file.md')
	expect(fileNameElement.nodeName).toBe('A')

	const destinationElement = getByText('Documents')
	expect(destinationElement.nodeName).toBe('A')

	expect(baseElement.innerHTML).toMatchSnapshot()
})

test('Display correct information for creations', async() => {
	const { baseElement, getByText } = render(Activity, { props: { activity: new ActivityModel(wsData.ocs.data[17]) } })

	expect(baseElement.textContent).toContain('You created  Test file.md')

	const fileNameElement = getByText('Test file.md')
	expect(fileNameElement.nodeName).toBe('A')

	expect(baseElement.innerHTML).toMatchSnapshot()
})
