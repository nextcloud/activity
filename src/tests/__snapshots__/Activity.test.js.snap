// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display correct information for changes 1`] = `
<div>
  <li class="activity-entry">
    <div data-v-69abe502="" class="avatardiv popovermenu-wrapper monochrome activity-entry__icon activity-icon" style="--size: 15px; line-height: 15px; font-size: 7px;">
      <!---->
      <!---->
      <!---->
      <!---->
    </div>
    <div class="activity-entry__content">
      <div data-v-a8ade67f="" class="activity-entry__content__subject rich-text--wrapper">
        <div data-v-a8ade67f="">You changed <span data-v-a8ade67f="" class="rich-text-file rich-text--component"><!----> <a title="" aria-label="" href="http://localhost:8080/index.php/f/776">Test file - renamed.md</a> <!----></span></div>
      </div>
      <div data-v-a8ade67f="" class="activity-entry__content__message rich-text--wrapper">
        <div data-v-a8ade67f=""></div>
      </div>
    </div> <span class="hidden-visually">April 29, 2021 10:51 AM</span> <span title="April 29, 2021 10:51 AM" class="activity-entry__date">11 days ago</span>
  </li>
</div>
`;

exports[`Display correct information for comments 1`] = `
<div>
  <li class="activity-entry">
    <div data-v-69abe502="" class="avatardiv popovermenu-wrapper monochrome activity-entry__icon activity-icon" style="--size: 15px; line-height: 15px; font-size: 7px;">
      <!---->
      <!---->
      <!---->
      <!---->
    </div>
    <div class="activity-entry__content">
      <div data-v-a8ade67f="" class="activity-entry__content__subject rich-text--wrapper">
        <div data-v-a8ade67f="">You commented</div>
      </div>
      <div data-v-a8ade67f="" class="activity-entry__content__message rich-text--wrapper">
        <div data-v-a8ade67f="">A comment</div>
      </div>
    </div> <span class="hidden-visually">April 29, 2021 11:04 AM</span> <span title="April 29, 2021 11:04 AM" class="activity-entry__date">11 days ago</span>
  </li>
</div>
`;

exports[`Display correct information for creations 1`] = `
<div>
  <li class="activity-entry">
    <div data-v-69abe502="" class="avatardiv popovermenu-wrapper activity-entry__icon activity-icon" style="--size: 15px; line-height: 15px; font-size: 7px;">
      <!---->
      <!---->
      <!---->
      <!---->
    </div>
    <div class="activity-entry__content">
      <div data-v-a8ade67f="" class="activity-entry__content__subject rich-text--wrapper">
        <div data-v-a8ade67f="">You created <span data-v-a8ade67f="" class="rich-text-file rich-text--component"><!----> <a title="" aria-label="" href="http://localhost:8080/index.php/f/776">Test file.md</a> <!----></span></div>
      </div>
      <div data-v-a8ade67f="" class="activity-entry__content__message rich-text--wrapper">
        <div data-v-a8ade67f=""></div>
      </div>
    </div> <span class="hidden-visually">April 29, 2021 10:49 AM</span> <span title="April 29, 2021 10:49 AM" class="activity-entry__date">11 days ago</span>
  </li>
</div>
`;

exports[`Display correct information for favorites 1`] = `
<div>
  <li class="activity-entry">
    <div data-v-69abe502="" class="avatardiv popovermenu-wrapper activity-entry__icon activity-icon" style="--size: 15px; line-height: 15px; font-size: 7px;">
      <!---->
      <!---->
      <!---->
      <!---->
    </div>
    <div class="activity-entry__content">
      <div data-v-a8ade67f="" class="activity-entry__content__subject rich-text--wrapper">
        <div data-v-a8ade67f="">Added to favorites</div>
      </div>
      <div data-v-a8ade67f="" class="activity-entry__content__message rich-text--wrapper">
        <div data-v-a8ade67f=""></div>
      </div>
    </div> <span class="hidden-visually">April 29, 2021 10:52 AM</span> <span title="April 29, 2021 10:52 AM" class="activity-entry__date">11 days ago</span>
  </li>
</div>
`;

exports[`Display correct information for moves 1`] = `
<div>
  <li class="activity-entry">
    <div data-v-69abe502="" class="avatardiv popovermenu-wrapper monochrome activity-entry__icon activity-icon" style="--size: 15px; line-height: 15px; font-size: 7px;">
      <!---->
      <!---->
      <!---->
      <!---->
    </div>
    <div class="activity-entry__content">
      <div data-v-a8ade67f="" class="activity-entry__content__subject rich-text--wrapper">
        <div data-v-a8ade67f="">You moved <span data-v-a8ade67f="" class="rich-text-file rich-text--component"><!----> <a title="" aria-label="" href="http://localhost:8080/index.php/f/776">Test file.md</a> <!----></span> to <span data-v-a8ade67f="" class="rich-text-file rich-text--component"><!----> <a title="" aria-label="" href="http://localhost:8080/index.php/f/753">Documents</a> <!----></span></div>
      </div>
      <div data-v-a8ade67f="" class="activity-entry__content__message rich-text--wrapper">
        <div data-v-a8ade67f=""></div>
      </div>
    </div> <span class="hidden-visually">April 29, 2021 10:49 AM</span> <span title="April 29, 2021 10:49 AM" class="activity-entry__date">11 days ago</span>
  </li>
</div>
`;

exports[`Display correct information for renames 1`] = `
<div>
  <li class="activity-entry">
    <div data-v-69abe502="" class="avatardiv popovermenu-wrapper monochrome activity-entry__icon activity-icon" style="--size: 15px; line-height: 15px; font-size: 7px;">
      <!---->
      <!---->
      <!---->
      <!---->
    </div>
    <div class="activity-entry__content">
      <div data-v-a8ade67f="" class="activity-entry__content__subject rich-text--wrapper">
        <div data-v-a8ade67f="">You renamed <span data-v-a8ade67f="" class="rich-text-file rich-text--component"><!----> <a title="" aria-label="" href="http://localhost:8080/index.php/f/776">Test file - renamed.md</a> <!----></span> to <span data-v-a8ade67f="" class="rich-text-file rich-text--component"><!----> <a title="" aria-label="" href="http://localhost:8080/index.php/f/776">Test file - renamed - looooooooong.md</a> <!----></span></div>
      </div>
      <div data-v-a8ade67f="" class="activity-entry__content__message rich-text--wrapper">
        <div data-v-a8ade67f=""></div>
      </div>
    </div> <span class="hidden-visually">April 29, 2021 11:12 AM</span> <span title="April 29, 2021 11:12 AM" class="activity-entry__date">11 days ago</span>
  </li>
</div>
`;

exports[`Display correct information for shares 1`] = `
<div>
  <li class="activity-entry">
    <div data-v-69abe502="" class="avatardiv popovermenu-wrapper monochrome activity-entry__icon activity-icon" style="--size: 15px; line-height: 15px; font-size: 7px;">
      <!---->
      <!---->
      <!---->
      <!---->
    </div>
    <div class="activity-entry__content">
      <div data-v-a8ade67f="" class="activity-entry__content__subject rich-text--wrapper">
        <div data-v-a8ade67f="">Shared as public link</div>
      </div>
      <div data-v-a8ade67f="" class="activity-entry__content__message rich-text--wrapper">
        <div data-v-a8ade67f=""></div>
      </div>
    </div> <span class="hidden-visually">April 29, 2021 10:49 AM</span> <span title="April 29, 2021 10:49 AM" class="activity-entry__date">11 days ago</span>
  </li>
</div>
`;

exports[`Display correct information for tags 1`] = `
<div>
  <li class="activity-entry">
    <div data-v-69abe502="" class="avatardiv popovermenu-wrapper monochrome activity-entry__icon activity-icon" style="--size: 15px; line-height: 15px; font-size: 7px;">
      <!---->
      <!---->
      <!---->
      <!---->
    </div>
    <div class="activity-entry__content">
      <div data-v-a8ade67f="" class="activity-entry__content__subject rich-text--wrapper">
        <div data-v-a8ade67f="">Added system tag <strong data-v-a8ade67f="" class="rich-text--component">tag1</strong></div>
      </div>
      <div data-v-a8ade67f="" class="activity-entry__content__message rich-text--wrapper">
        <div data-v-a8ade67f=""></div>
      </div>
    </div> <span class="hidden-visually">April 29, 2021 10:50 AM</span> <span title="April 29, 2021 10:50 AM" class="activity-entry__date">11 days ago</span>
  </li>
</div>
`;

exports[`Display correct information for unfavorites 1`] = `
<div>
  <li class="activity-entry">
    <div data-v-69abe502="" class="avatardiv popovermenu-wrapper monochrome activity-entry__icon activity-icon" style="--size: 15px; line-height: 15px; font-size: 7px;">
      <!---->
      <!---->
      <!---->
      <!---->
    </div>
    <div class="activity-entry__content">
      <div data-v-a8ade67f="" class="activity-entry__content__subject rich-text--wrapper">
        <div data-v-a8ade67f="">Removed from favorites</div>
      </div>
      <div data-v-a8ade67f="" class="activity-entry__content__message rich-text--wrapper">
        <div data-v-a8ade67f=""></div>
      </div>
    </div> <span class="hidden-visually">April 29, 2021 10:52 AM</span> <span title="April 29, 2021 10:52 AM" class="activity-entry__date">11 days ago</span>
  </li>
</div>
`;
